compiler/AST/foralls.cpp

```diff
@@ -462,6 +456,45 @@ buildFollowLoop(VarSymbol* iter,
   return followBlock;
 }
 
+//
+// Handle the case where the leader iterator is _iterator_for_loopexpr.
+// Not doing so confuses ReturnByRef and lowering of ForallStmts.
+//
+// Tests:
+//   library/packages/Collection/CollectionCounter.chpl
+//   library/standard/Random/deitz/test1D2D.chpl
+//   reductions/deitz/test_maxloc_reduce_wmikanik_bug2.chpl
```

Comment:
Is this a workaround for not using "new style" forall representation for certain loop expressions?